lua/plenary/filetype.lua

```diff
@@ -73,11 +73,23 @@ filetype._parse_shebang = function(head)
   return ''
 end
 
+local done_adding = false
 filetype.detect_from_extension = function(filepath)
   local ext = filetype._get_extension(filepath)
   ext = ext and ext:lower()
   local match = ext and filetype_table.extension[ext]
   if match then return match end
+  if not done_adding then
+    local tmp_filetypes = vim.fn.getcompletion('', 'filetype')
+    for _, v in ipairs(tmp_filetypes) do
+      if filetype_table.extension[v] == nil then
+        filetype_table.extension[v] = v
+      end
+    end
+    done_adding = true
+    match = ext and filetype_table.extension[ext]
+    if match then return match end
+  end
```

Comment:
can you pull this out to a function that makes it clear what is happening (just name it something good).

Then we can do this. it seems fine to me. 

The one other thing would that you should wrap the function call in `pcall` in case this is called within a tight lua loop, since that would error with `vim.fn`.

Reply:
Good point. Thanks :) Updated it